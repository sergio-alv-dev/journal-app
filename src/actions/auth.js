import { types } from "../types/types";
import { firebase, googleAuthProvider } from "../firebase/firebaseConfig";
import { uiFinishLoading, uiStartLoading } from "./ui";
import Swal from 'sweetalert2';
import { notesLogout } from "./notes";

export const startLoginEmailPassword = (email, password) => {
    
    
    return ( dispatch ) => {
        
        dispatch( uiStartLoading() );

         return firebase.auth().signInWithEmailAndPassword( email, password )
            .then( ({ user }) => {
                // console.log(user);
                dispatch( login(user.uid, user.displayName ) );
                dispatch( uiFinishLoading() );
            })
            .catch( e => {
                console.log(e);
                dispatch( uiFinishLoading() );
                Swal.fire('Error', e.message, 'error');
            })
        

    }
}

export const startRegisterWithEmailPasswordName = (email, password, name) => {
    return( dispatch ) => {
        firebase.auth().createUserWithEmailAndPassword( email, password )
        .then( async ({ user }) => {
            
            await user.updateProfile({ displayName:name });
            console.log(user);
            dispatch(
                login(user.uid, user.displayName )
            )
            
        })
        .catch( e => {
            console.log(e)
            Swal.fire('Error', e.message, 'error');
        })
    }

}

export const startGoogleLogin = () => {
    return ( dispatch ) => {

        // console.log( 'google' );

        firebase.auth().signInWithPopup( googleAuthProvider )
            .then( ({ user }) => {
                dispatch(
                    login(user.uid, user.displayName )
                )
                // console.log( userCred );
            })
            .catch( e => {
                console.log(e)
                Swal.fire('Error', e.message, 'error');

            })

    }
}


export const login = ( uid, displayName) => {
    return{
        type: types.login,
        payload: {
            uid,
            displayName
        }
    }
}

export const startLogout = () => {
    return async ( dispatch) => {
        await firebase.auth().signOut();

        dispatch( logout() );
        dispatch( notesLogout() );
    }
}

export const logout = () => ({
    type: types.logout
})