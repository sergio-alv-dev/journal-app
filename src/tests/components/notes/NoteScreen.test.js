import React from 'react';
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import { NoteScreen } from '../../../components/notes/NoteScreen';
import { activeNote , startDeleting } from '../../../actions/notes';



jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn(),
    startDeleting: jest.fn(),
}));


const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

const initState = {
    auth: {
        uid: '123',
        name: 'sergio'
    },
    ui: {
        loading: false,
        msgError: null
    },
    notes:{
        notes: [],
        active: {
            id: 'abc',
            title: 'hola, test',
            body: 'mundo, test',
            date: 1,
        }
    }
};

let store = mockStore( initState );
store.dispatch = jest.fn();

const wrapper = mount(

    <Provider store = { store }>
        <MemoryRouter>
            <NoteScreen />
        </MemoryRouter>
    </Provider>
);

describe('Pruebas en <NoteScreen />', () => {

    test('debe de mostrarse correctamente', () => { expect( wrapper ).toMatchSnapshot()});
    test('debe de disparar el activeNote', () => {

        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'hola otra vez'
            }
        });

        expect( activeNote ).toHaveBeenCalledWith( 
            initState.notes.active.id,
            {
                id: 'abc',
                title: 'hola otra vez',
                body:'mundo, test',
                date: 1
            }
        );
        
        // expect( activeNote ).toHaveBeenCalledWith(
        //     1234,
        //     {
        //         id: 'abc',
        //         title: 'hola otra vez',
        //         body:'mundo, test',
        //         date: 0
        //     }
        // )

    })
    
    
    
})
