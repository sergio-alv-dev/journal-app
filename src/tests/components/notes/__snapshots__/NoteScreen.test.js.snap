// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NoteScreen /> debe de mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className=" notes__appbar"
  >
    <span>
       26 de feb 1992
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className=" notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="The best Title"
      type="text"
      value="hola, test"
    />
    <textarea
      className="notes__text-area"
      name="body"
      onChange={[Function]}
      placeholder="What happend today ?"
      value="mundo, test"
    />
  </div>
  <button
    className=" btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
